tests/spicy/types/unit/skip-field-fail.spicy

```diff
@@ -0,0 +1,16 @@
+# @TEST-EXEC-FAIL: spicyc -p %INPUT >> output 2>&1
+# @TEST-EXEC: btest-diff output
+
+# @TEST-DOC: Validation for `skip` fields.
+
+module Test;
+
+type X = unit {
+    a: skip bytes &size=1 { print $$; }
+    b: skip bytes &size=1 &requires=($$ == b"x");
+    c: skip bytes &size=1 &convert=$$;
+};
+
+on X::a {
+    print $$;
+}
```

Comment:
For the following grammar

```ruby
module foo;

public type X = unit {
    b: skip bytes &eod { print self.b; }
};
```

I get the following very generic error message:

```
[error] /tmp/foo.spicy:5:22: type does not have field 'b'
[error] spicyc: aborting after errors
```

This makes sense since `b` is not really there, but still seems confusing to me. If it doesn't require too much work it would be nice if we could somehow teach the resolver to emit a different error message for `skip` fields, e.g., `cannot reference field 'b' since the field is a skip field`.